crates/handlers/src/passwords.rs

```diff
@@ -186,7 +230,7 @@ impl PasswordManager {
         scheme: SchemeVersion,
         password: Zeroizing<String>,
         hashed_password: String,
-    ) -> Result<Option<(SchemeVersion, String)>, anyhow::Error> {
+    ) -> Result<PasswordVerificationResult<Option<(SchemeVersion, String)>>, anyhow::Error> {
```

Comment:
Not the prettiest, but this could return `Result<bool, Option<(SchemaVersion, String)>>`. I don't think adding this new result type makes sense